test-cargo-miri/run-test.py

```diff
@@ -105,6 +105,18 @@ def test_cargo_miri_test():
         "test.subcrate.stdout.ref", "test.stderr-empty.ref",
         env={'MIRIFLAGS': "-Zmiri-disable-isolation"},
     )
+    test("`cargo miri test` (proc-macro-crate)",
+        cargo_miri("test") + ["-p", "proc-macro-crate"],
+        "test.proc-macro-crate.stdout.ref", "test.stderr-proc-macro.ref",
+    )
+    test("`cargo miri test` (subcrate and proc-macro-crate, lib target)",
+        cargo_miri("test") + ["-p", "subcrate", "-p", "proc-macro-crate", "lib"],
+        "test.subcrate-proc-macro-crate.lib-target.stdout.ref", "test.stderr-proc-macro.ref",
+    )
+    test("`cargo miri test` (proc-macro-crate, test target)",
+        cargo_miri("test") + ["-p", "proc-macro-crate", "--test", "test"],
+        "test.proc-macro-crate.test-target.stdout.ref", "test.stderr-empty.ref",
+    )
```

Comment:
I think just *skipping* unit tests does not need so many tests, but I borrowed these tests from #1677 directly.

Reply:
I reworked the test. Hope it looks better now.